### PR TITLE
Replace ctrlp with fzf

### PR DESCRIPTION
**Pull request by cassiobotaro**
Use fzf instead of ctrlp plugin. fzf is faster and has more resources. 
I choose --bin option to avoid system replaces(like ctrl+r).
Another decision is to not replace grep.vim as proposed in #244 because it only works if silver_archer is installed. If we found a solution for that, we can do the replacement.

### CODE REVIEW

File: vim_template/vimrc [hunk 1]
```diff
@@ -345,25 +345,20 @@ noremap <Leader>e :e <C-R>=expand("%:p:h") . "/" <CR>
...
 noremap <Leader>te :tabe <C-R>=expand("%:p:h") . "/" <CR>
 
-"" ctrlp.vim
+"" fzf.vim
 set wildmode=list:longest,list:full
 set wildignore+=*.o,*.obj,.git,*.rbc,*.pyc,__pycache__
-let g:ctrlp_custom_ignore = '\v[\/](node_modules|target|dist)|(\.(swp|tox|ico|git|hg|svn))$'
-let g:ctrlp_user_command = "find %s -type f | grep -Ev '"+ g:ctrlp_custom_ignore +"'"
-let g:ctrlp_use_caching = 1
+let $FZF_DEFAULT_COMMAND =  "find * -path '*/\.*' -prune -o -path 'node_modules/**' -prune -o -path 'target/**' -prune -o -path 'dist/**' -prune -o  -type f -print -o -type l -print 2> /dev/null"
```
Comment (avelino) [inline vim_template/vimrc]: does not ignore:
- .git
- .hg
- .svn
- .ico
- .swp

Comment (cassiobotaro) [inline vim_template/vimrc]: -path '*/\.*' -prune -o

Comment (cassiobotaro) [inline vim_template/vimrc]: obs: this command looks weird, but is the default command only changed to include dist,target and node_modules ignore

Comment (avelino) [inline vim_template/vimrc]: sorry

**Pull request merged**